colossalai/zero/sharded_optim/sharded_optim.py

```diff
@@ -146,7 +150,11 @@ def __init__(
 
             # create a copy of fp32 weights of the parameters for which this rank is responsible
             fp16_flat_current_rank = self._param_store.get_flat_fp16_param_by_rank_group(self._local_rank, group_id)
-            fp32_flat_current_rank = fp16_flat_current_rank.clone().float().detach()
+            # when using cpu offload, our cpu adam support fp16 paramters
+            if self._cpu_fp16_param:
+                fp32_flat_current_rank = fp16_flat_current_rank.clone().detach()
```

Comment:
Maybe you can just call `detach()`, which returns a new Tensor detached from the current graph. See https://pytorch.org/docs/stable/generated/torch.Tensor.detach.html?highlight=detach#torch.Tensor.detach

Reply:
ok